storage_scrubber/src/checks.rs

```diff
@@ -47,6 +48,39 @@ impl TimelineAnalysis {
     }
 }
 
+fn check_valid_layermap(metadata: &HashMap<LayerName, LayerFileMetadata>) -> Option<String> {
+    let mut lsn_split_point = BTreeSet::new(); // TODO: use a better data structure (range tree / range set?)
+    let mut all_delta_layers = Vec::new();
+    for (name, _) in metadata.iter() {
+        if let LayerName::Delta(layer) = name {
+            all_delta_layers.push(layer.clone());
+        }
+    }
+    for layer in &all_delta_layers {
+        if layer.key_range.start.next() != layer.key_range.end {
+            let lsn_range = &layer.lsn_range;
+            lsn_split_point.insert(lsn_range.start);
+            lsn_split_point.insert(lsn_range.end);
+        }
+    }
+    for layer in &all_delta_layers {
+        let key_range = &layer.key_range;
+        if key_range.start.next() != key_range.end {
```

Comment:
nit: we could just collect `layer.key_range.start.next() != layer.key_range.end` layers in `all_delta_layers` instead of checking the condition every time.